openss7-0.9.2.D.SLES9_K2.6.5_rc1(1)/openss7-0.9.2.D.SLES9_K2.6.5_rc1/striso/doc/man/man4/isot.4.man
'\" rtp
.\" -*- nroff -*- vim: ft=nroff nocin nosi
.\"
.\" @(#) isot.4.man,v 0.9.2.1 2006/04/11 18:22:13 brian Exp
.\"
.\" =========================================================================
.\"
.\" Copyright (c) 2001-2006  OpenSS7 Corporation <http://www.openss7.com/>
.\"
.\" All Rights Reserved.
.\"
.\" Permission is granted to make and distribute verbatim copies of this
.\" manual provided the copyright notice and this permission notice are
.\" preserved on all copies.
.\"
.\" Permission is granted to copy and distribute modified versions of this
.\" manual under the conditions for verbatim copying, provided that the
.\" entire resulting derived work is distributed under the terms of a
.\" permission notice identical to this one
.\" 
.\" Since the Linux kernel and libraries are constantly changing, this
.\" manual page may be incorrect or out-of-date.  The author(s) assume no
.\" responsibility for errors or omissions, or for damages resulting from
.\" the use of the information contained herein.  The author(s) may not
.\" have taken the same level of care in the production of this manual,
.\" which is licensed free of charge, as they might when working
.\" professionally.  The author(s) will take no responsibility in it.
.\" 
.\" Formatted or processed versions of this manual, if unaccompanied by
.\" the source, must acknowledge the copyright and authors of this work.
.\"
.\" -------------------------------------------------------------------------
.\"
.\" U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software
.\" on behalf of the U.S. Government ("Government"), the following
.\" provisions apply to you.  If the Software is supplied by the Department
.\" of Defense ("DoD"), it is classified as "Commercial Computer Software"
.\" under paragraph 252.227-7014 of the DoD Supplement to the Federal
.\" Acquisition Regulations ("DFARS") (or any successor regulations) and the
.\" Government is acquiring only the license rights granted herein (the
.\" license rights customarily provided to non-Government users).  If the
.\" Software is supplied to any unit or agency of the Government other than
.\" DoD, it is classified as "Restricted Computer Software" and the
.\" Government's rights in the Software are defined in paragraph 52.227-19
.\" of the Federal Acquisition Regulations ("FAR") (or any successor
.\" regulations) or, in the cases of NASA, in paragraph 18.52.227-86 of the
.\" NASA Supplement to the FAR (or any successor regulations).
.\"
.\" =========================================================================
.\" 
.\" Commercial licensing and support of this software is available from
.\" OpenSS7 Corporation at a fee.  See http://www.openss7.com/
.\" 
.\" =========================================================================
.\"
.\" Last Modified 2006/04/11 18:22:13 by brian
.\"
.\" -------------------------------------------------------------------------
.\"
.\" isot.4.man,v
.\" Revision 0.9.2.1  2006/04/11 18:22:13  brian
.\" - added manual pages and references
.\"
.\" =========================================================================
.R1
bracket-label "\fR[\fB" "\fR]" "\fR, \fB"
no-default-database
database striso.refs
accumulate
move-punctuation
abbreviate A
join-authors ", " ", " " and "
et-al " et al" 2 3
abbreviate-label-ranges ".."
sort-adjacent-labels
.R2
.so striso.macros
.\"
.\"
.TH ISOT 4 "@PACKAGE_DATE@" "@PACKAGE@-@VERSION@" "@PACKAGE_TITLE@ Devices"
.SH NAME
.B ISOT
\- ISO Transport over TCP (ISOT)
.SH DESCRIPTION
.B ISOT
is an implementation of ISO Transport over TCP (ISOT) as described in RFCs.
.[
rfc983
.]
.[
rfc1006
.]
The RFC describes the use of IS 8073
.[
ISO/IEC 8073
.]
ISO Transport Class 0 over TCP/IP (ISOT) by using TCP/IP as a Connection-oriented Network Service
Provider instead of X.25 CONS.
.[
ISO/IEC 8878
.]
.PP
The RFC describes adaptation of TCP to a CONS network service provider through
the use of a 4 byte header used to delimit NPDUs inside of the TCP stream.
.[
$LIST$
.]
.TI


openss7-0.9.2.D.SLES9_K2.6.5_rc1(1)/openss7-0.9.2.D.SLES9_K2.6.5_rc1/striso/src/modules/isot.c
/*****************************************************************************

 @(#) isot.c,v OpenSS7-0_9_2_D_rc2(1.1.2.1) 2006/07/11 12:32:05

 -----------------------------------------------------------------------------

 Copyright (c) 2001-2006  OpenSS7 Corporation <http://www.openss7.com/>
 Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>

 All Rights Reserved.

 This program is free software; you can redistribute it and/or modify it under
 the terms of the GNU General Public License as published by the Free Software
 Foundation; version 2 of the License.

 This program is distributed in the hope that it will be useful, but WITHOUT
 ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
 details.

 You should have received a copy of the GNU General Public License along with
 this program; if not, write to the Free Software Foundation, Inc., 675 Mass
 Ave, Cambridge, MA 02139, USA.

 -----------------------------------------------------------------------------

 U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software on
 behalf of the U.S. Government ("Government"), the following provisions apply
 to you.  If the Software is supplied by the Department of Defense ("DoD"), it
 is classified as "Commercial Computer Software" under paragraph 252.227-7014
 of the DoD Supplement to the Federal Acquisition Regulations ("DFARS") (or any
 successor regulations) and the Government is acquiring only the license rights
 granted herein (the license rights customarily provided to non-Government
 users).  If the Software is supplied to any unit or agency of the Government
 other than DoD, it is classified as "Restricted Computer Software" and the
 Government's rights in the Software are defined in paragraph 52.227-19 of the
 Federal Acquisition Regulations ("FAR") (or any successor regulations) or, in
 the cases of NASA, in paragraph 18.52.227-86 of the NASA Supplement to the FAR
 (or any successor regulations).

 -----------------------------------------------------------------------------

 Commercial licensing and support of this software is available from OpenSS7
 Corporation at a fee.  See http://www.openss7.com/

 -----------------------------------------------------------------------------

 Last Modified 2006/07/11 12:32:05 by brian

 -----------------------------------------------------------------------------

 isot.c,v
 Revision 1.1.2.1  2006/07/11 12:32:05  brian
 - added ISO and other implementations to distribution

 Revision 0.9.2.1  2006/04/11 18:30:10  brian
 - added new ISO over TCP modules

 *****************************************************************************/

#ident "@(#) isot.c,v OpenSS7-0_9_2_D_rc2(1.1.2.1) 2006/07/11 12:32:05"

static char const ident[] = "isot.c,v OpenSS7-0_9_2_D_rc2(1.1.2.1) 2006/07/11 12:32:05";

/*
 *  ISO Transport over TCP/IP (ISOT)
 *
 *  ISOT implements ISO Transport Protocol Class 0 (TP0) over TCP/IP in
 *  accordance with RFC 1006.  This module implements the upper layer protocol
 *  module that constitutes a specialized ISO TS-Provider that implements the
 *  ISOT protocol to TCP acting as an NS-Provider.  This module presents an
 *  ISO TPI interface to the TS-User above it.
 *
 *  This module is meant to be pushed over a TCP Transport Provider Interface
 *  (TPI) stream that already has a "tcpns" module pushed over it.
 */
